### PR TITLE
Add minimum time flag for processlist metrics

### PR DESCRIPTION
**Pull request by SuperQ**
Add a flag for controlling the minimum time that a thread must be in a
specific state in order to be counted.

This allows for detection of threads "stuck" in a specific state for a
minimum period of time.  This avoids counting short-lived thread state
chagnes from polluting the metrics.

### CODE REVIEW

**Comment by SuperQ** [comment]
@rtreffer  This helps make the processlist metric more useful to find long-running queries.

**Comment by SuperQ** [comment]
@juliusv :-D

File: mysqld_exporter.go [hunk 1]
```diff
@@ -40,6 +40,10 @@ var (
 		"collect.info_schema.processlist", false,
 		"Collect current thread state counts from the information_schema.processlist",
 	)
+	processlistMinTime = flag.Int(
+		"collect.info_schema.processlist.min_time", 0,
+		"Minimum time a thread must be in each state to be counted",
```
Comment (juliusv) [inline mysqld_exporter.go]: nit: some of the HELP strings have periods while others don't.

**Comment by juliusv** [comment]
another nit: s/minium/minimum/ in PR/commit title.

**Comment by juliusv** [comment]
:+1: otherwise

File: mysqld_exporter.go (hunk 1 above)
Comment (SuperQ) [inline mysqld_exporter.go]: I'll fix that separately.

**Comment by juliusv** [comment]
:+1:

**Pull request merged**